custom/icds_reports/utils/aggregation_helpers/distributed/location_reassignment.py

```diff
@@ -0,0 +1,174 @@
+from dateutil.relativedelta import relativedelta
+from django.db import connections
+
+from corehq.apps.userreports.util import get_table_name
+from corehq.sql_db.connections import get_icds_ucr_citus_db_alias
+from custom.icds_reports.const import DASHBOARD_DOMAIN
+from custom.icds_reports.utils.aggregation_helpers import get_prev_agg_tablename, transform_day_to_month
+
+
+class TempPrevTablesBase(object):
+
+    DROP_QUERY = """
+    DROP TABLE IF EXISTS "{prev_table}";
+    DROP TABLE IF EXISTS "{prev_local}";
+    """
+
+    def drop_temp_tables(self, alias):
+        data = {
+            'prev_table': get_prev_agg_tablename(alias),
+            'prev_local': f"{alias}_prev_local",
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.DROP_QUERY.format(**data))
+
+    def create_temp_tables(self, alias, table, day):
+        data = {
+            'prev_table': get_prev_agg_tablename(alias),
+            'prev_local': f"{alias}_prev_local",
+            'prev_month': day,
+            'current_table': table,
+            'alias': alias
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.CREATE_QUERY.format(**data))
+
+    def make_all_tables(self, day):
+        raise NotImplementedError
+
+
+class TempPrevUCRTables(TempPrevTablesBase):
+
+    CREATE_QUERY = """
+    CREATE UNLOGGED TABLE "{prev_table}" (LIKE "{current_table}");
```

Comment:
should it also include `INCLUDE CONSTRAINS INDEXES`?

Reply:
I did not include those on purpose, since they slow down the insert. I will add ones that are used by the agg, but do in separately at the end so that insert is fast and we don't have unnecessary indices